### PR TITLE
Extract domain entities

### LINKED ISSUE #201: Extract last domain entities from graphql module
During the restructuring of the proxy in #125 - a couple of entities were left in the graphql module and are in need of rescuing. Additionally #199 also introduced some of these and would run out of scope if done in there. We should extract:

* Project
* Identity

### PR DESCRIPTION
**Pull request by xla**
Closes #201

### CODE REVIEW

File: proxy/src/error.rs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-//! Domain errors returned by the API.
+//! Domain errors.
```
Comment (NunoAlexandre) [inline proxy/src/error.rs]: Just "Domain errors" seems lacking to me.

File: proxy/src/project.rs [hunk 2]
```diff
@@ -1,27 +1,12 @@
-//! Abstractions for projects returned by the API.
+//! Abstractions for projectsI.
```
Comment (NunoAlexandre) [inline proxy/src/project.rs]:
```suggestion
//! Abstractions for projects.
```

Could you expand that comment? What abstractions? Why do we need them?

**Comment by xla** [comment]
@rudolfs Thanks for the catch, why is this not surfaced by our tests? Is it because of #159 ?

**Comment by rudolfs** [comment]
@xla yes, #159, #145 and #146.

**Comment by xla** [comment]
> Looks good! `ProjectMetadata` would be enough for me instead of `ProjectMetadataInput` but leave that with you +1

@NunoAlexandre `ProjectMetadata` exists as well. GraphQL distinguishes between input values and returned values. So input to queries and mutations are expressed differently. Therefore the on the surface superfluous looking suffix.

**Comment by NunoAlexandre** [comment]
@xla that's clear! And, In that case, I like the suffix to make things explicit. Thanks for explaining!

**Pull request merged**